app/main/views/users.py

```diff
@@ -84,6 +95,10 @@ def auth_user():
         db.session.add(audit)
         db.session.commit()
 
+        if user.locked:
+            # Hash the locked user's password and ignore the result, to mitigate against timing attacks
+            encryption.checkpw("not a real password", user.password)
```

Comment:
> the hashing should get performed for locked users anyway, no?

That's what I thought, but looking inside `encryption.authenticate_user` reveals otherwise.

Reply:
> causing this to get executed again
That is a good point - I'll shift the code up to before we increment the count.

Reply:
> That's what I thought, but looking inside `encryption.authenticate_user` reveals otherwise.

Oh - shouldn't we just be fixing `encryption.authenticate_user` then to switch the predicates around? It seems like that's the function that's "wrong" and introducing cases where it'll do the hashing in one case and not the other...

Reply:
Fair. I'll see what I can do.

Reply:
Added a commit (and a test).